packages/roosterjs-editor-core/lib/coreApi/insertNode.ts

```diff
@@ -179,3 +178,23 @@ export const insertNode: InsertNode = (core: EditorCore, node: Node, option: Ins
 
     return true;
 };
+function adjustInsertPositionNewLine(blockElement: BlockElement, core: EditorCore, pos: Position) {
+    let tempPos = new Position(blockElement.getEndNode(), PositionType.After);
+    if (safeInstanceOf(tempPos.node, 'HTMLTableRowElement')) {
+        const div = core.contentDiv.ownerDocument.createElement('div');
+        div.id = 'newLineInTd';
+        insertNode(core, div, {
+            position: ContentPosition.SelectionStart,
+            insertOnNewLine: false,
+            updateCursor: true,
+            replaceSelection: true,
+        });
```

Comment:
Inside insertNode() function, can we avoid calling it recursively? Can we just directly insert the node using some DOM API such as insertBefore or appendChild?

Reply:
Sounds good will use the Range.insertNode Api to do this.